packages/manager/modules/pci/src/projects/project/kubernetes/details/service/OidcProvider.class.js

```diff
@@ -0,0 +1,18 @@
+export default class OidcProvider {
+  /**
+   * Create a OIDC Provider instance
+   * @param oidcProviderModel {Object}: native oidcProvider JS object
+   * */
+  constructor(oidcProviderModel) {
+    Object.assign(this, oidcProviderModel);
+  }
+
+  isDefined() {
+    const { clientId, issuerUrl } = this;
+    return clientId && clientId !== '' && issuerUrl && issuerUrl !== '';
+  }
+
+  copy() {
+    return new OidcProvider(this);
```

Comment:
if you are trying to clone the object, you can use return Object.assign({}, this)

Reply:
Currently the OidcProvider constructor do that.
It copy only properties.